### ISSUE TITLE
Van der Pol benchmark

### ISSUE THREAD

**Issue by mforets** [JuliaReach/ReachabilityBenchmarks#84]
Using benchmark  / property checking mode settings:

```julia
julia> include("models/ARCH/NLN/VanDerPol/vanderpol_benchmark.jl")
(1/1) benchmarking "VanDerPol"...
  (1/1) benchmarking "x[2] <= 2.75"...
  done (took 5.640603636 seconds)
done (took 6.286683375 seconds)
minimum time for each benchmark:
1-element BenchmarkTools.BenchmarkGroup:
  tags: []
  "VanDerPol" => 1-element BenchmarkTools.BenchmarkGroup:
          tags: []
          "x[2] <= 2.75" => TrialEstimate(96.674 ms)
median time for each benchmark:
1-element BenchmarkTools.BenchmarkGroup:
  tags: []

  "VanDerPol" => 1-element BenchmarkTools.BenchmarkGroup:
          tags: []
          "x[2] <= 2.75" => TrialEstimate(109.770 ms)
```

<img width="581" alt="screen shot 2019-02-28 at 17 38 55" src="https://user-images.githubusercontent.com/12424594/53597921-212e9900-3b82-11e9-80e4-d546d6e2f1a7.png">

Using default settings:

```julia
julia> @btime vanderpol_TMJets(; t0=0.0, T=7.0, abs_tol=1e-20, orderT=13, orderQ=2,
                                   maxsteps=500, property=x->x[2] <= 2.75);
  1.056 s (17631756 allocations: 955.92 MiB)
```

```julia
julia> @btime vanderpol_TMJets(; t0=0.0, T=7.0, abs_tol=1e-20, orderT=13, orderQ=2,
                                   maxsteps=500, property=x->x[2] <= 2.75);
  1.056 s (17631756 allocations: 955.92 MiB)
```

<img width="577" alt="screen shot 2019-02-28 at 17 28 55" src="https://user-images.githubusercontent.com/12424594/53597970-46230c00-3b82-11e9-8eb4-3e33b8799770.png">

Again i think that having fast property checking settings and accurate default settings is very cool.

CC: @lbenet

**Comment by schillic** [comment]
This is so cool! You can use this tool as a generator for modern arts :art: :framed_picture: :moneybag:

**Comment by mforets** [comment]
Update for benchmark settings:

```julia
julia> include("models/ARCH/NLN/VanDerPol/vanderpol_benchmark.jl")
[ Info: Recompiling stale cache file /Users/forets/.julia/compiled/v1.1/Reachability/CHV4V.ji for Reachability [774a0091-654f-5c65-bbdc-ad5b67b45832]
(1/1) benchmarking "VanDerPol"...
  (1/1) benchmarking "x[2] <= 2.75"...
  done (took 5.637534728 seconds)
done (took 6.182533775 seconds)
minimum time for each benchmark:
1-element BenchmarkTools.BenchmarkGroup:
  tags: []
  "VanDerPol" => 1-element BenchmarkTools.BenchmarkGroup:

          tags: []
          "x[2] <= 2.75" => TrialEstimate(170.446 ms)
median time for each benchmark:
1-element BenchmarkTools.BenchmarkGroup:
  tags: []
  "VanDerPol" => 1-element BenchmarkTools.BenchmarkGroup:
          tags: []
          "x[2] <= 2.75" => TrialEstimate(201.087 ms)
```

**Comment by lbenet** [comment]
So, the benchmarks haven't changed that much, or have they?

**Comment by mforets** [comment]
It came ~2x slower, but still well under 1second. Note that the *median* of the samples went up to 200ms, while before the update it was at 100ms, this is the time argument of the `TrialEstimate` shown above. 

(I show `minimum` and `median` because they are better estimates than the mean; see the discussion in [benchmarktools](https://github.com/JuliaCI/BenchmarkTools.jl/blob/master/doc/manual.md#which-estimator-should-i-use).)

**Comment by mforets** [comment]
Update for the latest commits in [TaylorModels](https://github.com/JuliaIntervals/TaylorModels.jl/pull/18): the benchmark time remains unchanged, even slightly faster, and the property is verified.

```julia
# benchmark settings
𝑂 = Options(:t0=>0.0, :T=>7.0, :abs_tol=>1e-1, :orderT=>2, :orderQ=>2,
            :maxsteps=>500, :property=>(t, x) -> x[2] < 2.75)

julia> include("models/ARCH/NLN/VanDerPol/vanderpol_benchmark.jl")
(1/1) benchmarking "VanDerPol"...
  (1/1) benchmarking "x[2] <= 2.75"...
  done (took 5.731805599 seconds)
done (took 6.311149062 seconds)
minimum time for each benchmark:
1-element BenchmarkTools.BenchmarkGroup:
  tags: []
  "VanDerPol" => 1-element BenchmarkTools.BenchmarkGroup:
          tags: []
          "x[2] <= 2.75" => TrialEstimate(158.764 ms)
median time for each benchmark:
1-element BenchmarkTools.BenchmarkGroup:
  tags: []

  "VanDerPol" => 1-element BenchmarkTools.BenchmarkGroup:
          tags: []
          "x[2] <= 2.75" => TrialEstimate(168.457 ms)
```

Another improvement  👍  is that the fast (ie. benchmark) settings are also more accurate than before:

<img width="589" alt="Screen Shot 2019-03-08 at 17 49 17" src="https://user-images.githubusercontent.com/12424594/54055022-87429e00-41ca-11e9-82e5-7f22104e0188.png">

On the other hand, the default options (runs in ~1.6sec) give

![Screen Shot 2019-03-08 at 17 50 13](https://user-images.githubusercontent.com/12424594/54055111-bb1dc380-41ca-11e9-8a16-af582fe46bf5.png)

**Comment by lbenet** [comment]
Maybe the following parameters are better than the current default: `abs_tol = 1.0e-10`, `orderT = 10`, `orderQ = 2`. I get running times ~ 0.6 sec.

**Comment by mforets** [comment]
That's good. I've updated the default. In terms of accuracy, by "eye" inspection seems to be rather close to the one with benchmark settings, see picture below.

```julia
julia> @btime vanderpol_TMJets();
  530.975 ms (8997451 allocations: 510.32 MiB)
```

<img width="585" alt="Screen Shot 2019-03-09 at 10 17 52" src="https://user-images.githubusercontent.com/12424594/54072095-c61d3600-4254-11e9-9814-ad866ac2d9a5.png">

**Comment by schillic** [comment]
We can verify the scenario with μ = 2 in 7 seconds by splitting the initial states uniformly into three rectangles in the x1 dimension. I did not play with the parameters.

**Comment by mforets** [comment]
Final plots and runtimes after having merged #131:

```julia
julia> include("models/ARCH/NLN/VanDerPol/vanderpol_benchmark.jl")
(1/1) benchmarking "VanDerPol"...

  (1/2) benchmarking "μ = 2: x[2] <= 4.0"...
  done (took 6.578803873 seconds)
  (2/2) benchmarking "μ = 1: x[2] <= 2.75"...
  done (took 5.787473946 seconds)
done (took 12.942305166 seconds)
minimum time for each benchmark:
1-element BenchmarkTools.BenchmarkGroup:
  tags: []
  "VanDerPol" => 2-element BenchmarkTools.BenchmarkGroup:
          tags: []
          "μ = 2: x[2] <= 4.0" => TrialEstimate(5.963 s)
          "μ = 1: x[2] <= 2.75" => TrialEstimate(541.583 ms)
median time for each benchmark:
1-element BenchmarkTools.BenchmarkGroup:
  tags: []
  "VanDerPol" => 2-element BenchmarkTools.BenchmarkGroup:
          tags: []
          "μ = 2: x[2] <= 4.0" => TrialEstimate(5.963 s)
          "μ = 1: x[2] <= 2.75" => TrialEstimate(560.198 ms)
```

![vanderpol_case_1](https://user-images.githubusercontent.com/12424594/55267518-6cab9400-5261-11e9-9472-0e73b7dbffc3.png)


![vanderpol_case_2](https://user-images.githubusercontent.com/12424594/55267522-71704800-5261-11e9-8d8e-d969f2effe08.png)

![vanderpol_case_all](https://user-images.githubusercontent.com/12424594/55267534-73d2a200-5261-11e9-9dc1-7b28af9a21e5.png)
